packages/nuxt/src/core/app.ts

```diff
@@ -194,18 +197,14 @@ export async function resolveApp (nuxt: Nuxt, app: NuxtApp) {
       }
       app.middleware.push({ name, path: file, global: hasSuffix(file, '.global') })
     }
-  }
-
-  // Resolve plugins, first extended layers and then base
-  app.plugins = []
-  for (const config of reversedConfigs) {
-    const pluginDir = (config.rootDir === nuxt.options.rootDir ? nuxt.options : config).dir?.plugins || 'plugins'
+    // Resolve plugins, first extended layers and then base
+    const pluginDir = baseDir?.plugins || 'plugins'
     app.plugins.push(...[
       ...(config.plugins || []),
       ...config.srcDir
         ? await resolveFiles(config.srcDir, [
-          `${pluginDir}/*{${nuxt.options.extensions.join(',')}}`,
-          `${pluginDir}/*/index{${nuxt.options.extensions.join(',')}}`,
+          `${pluginDir}/*{${extensions}}`,
+          `${pluginDir}/*/index{${extensions}}`,
         ])
         : [],
     ].map(plugin => normalizePlugin(plugin as NuxtPlugin)))
```

Comment:
Question: in L222-L223, is that even necessary? Can we not call them just once in L238-L239?